hybrasyl/Objects/Monster.cs

```diff
@@ -146,6 +531,148 @@ public void Sleep()
             //return to idle state
         }
 
+
+        public override bool Walk(Direction direction)
+        {
```

Comment:
Why all the duplication here? `VisibleObjects` can and should be able to `Walk`, rather than duplicating this from players.

Reply:
An Item is a VisibleObject. Should an Item be able to walk?

Reply:
Yes, some of them have feet.

Reply:
This may be viable at a creature level, however, both user and monster would have to use override methods as is due to differences in the method contents.

Reply:
I'd rather have override methods than all this duplication. In particular, the viewport calculation can be removed to a shared function in `Creature`.

Reply:
The viewport calculation could be an override, however, in both cases of player and monster, the code that currently exists will remain. You can dislike this all you'd like, however, it's not 100% identical between the two, and thus, cannot be made generic.

I'm not going to start refactoring out parts of methods into common methods unless it 100% warrants it. In this case, it does not. It's just a waste of time.